### PR TITLE
Improve Wake rules for formerly Wit Scala plugin

### PR DESCRIPTION
**Pull request by jackkoenig**
Simplify the file ownership so that fetched Scala dependencies are owned
by a value that always runs and there is no differentiation between
files fetched by Wit and by Wake. This should improve backwards and
forwards Wit compatibility.

This also makes the Scala fetching job a value instead of a function so
it always runs which provides preinstall-like behavior by simply
invoking `wake`.

This is primarily just a simplification, rather than trying to differentiate between "wit fetched" and "my fetched" Scala stuff, we just use a LocalOnly job and own all of the files, regardless of who fetched them.

Improvements:
* Fixes the issue in https://github.com/sifive/api-scala-sifive/pull/27#issuecomment-607065736
* Supports preinstall-like behavior by merely running `wake`
* Supports preinstall from existing Wit Scala fetch (ie. without having to run `wake` if you use Wit with scala plugin)

TODO:
- [x] Fix tests (lol)
- [x] Test of preinstall-like behavior

### CODE REVIEW

**Comment by jackkoenig** [comment]
I think this is sufficient to crack `v0.2.0` and maintains old Wit Scala plugin behavior. Then I can work on `v0.3.0` which will have incremental compile (and probably continue backwards compatibility with Wit although will require explicit `wake` to fetch all Scala stuff)

**Comment by jackkoenig** [comment]
@richardxia I tweaked it slightly, I made the `ivydependencies.json` sources include an `ivydependencies.json` at the workspace root which fixes the tests. It is a slight change so I wanted to highlight it.

File: rerun.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#! /bin/sh -ex
```
Comment (richardxia) [inline rerun.sh:1]: I assume you didn't want to commit this file?

Comment (jackkoenig) [reply rerun.sh:1]: Lol nope, will fix

**Pull request merged**